Data/ByteString/Internal.hs

```diff
@@ -648,6 +649,25 @@ concat = \bss0 -> goLen0 bss0 bss0
    concat [x] = x
  #-}
 
+-- | /O(log n)/ Repeats the given ByteString n times.
+times :: Integral a => a -> ByteString -> ByteString
+times n (BS fp len)
+  | n < 0 = error "stimes: non-negative multiplier expected"
+  | n == 0 = mempty
+  | otherwise = unsafeCreate size $ \destptr ->
+    withForeignPtr fp $ \p -> do
+      memcpy destptr p len
+      fillFrom destptr len
```

Comment:
Did you consider optimizing the cases where the input bytestring has length 0 or 1? Not sure whether it's worth adding more complexity though.

Reply:
Just optimized it for the 0 and 1 cases of len and n